src/generic/SparsePoly.jl

```diff
@@ -478,13 +478,13 @@ function evaluate(a::SparsePoly{T}, b::S) where {S <: RingElement, T <: RingElem
    if a.length == 0
       return base_ring(a)()
    end
-   r = a.coeffs[a.length]
+   r = deepcopy(a.coeffs[a.length])
    for i = 1:a.length - 1
-      r *= b^(reinterpret(Int, a.exps[a.length - i + 1] - a.exps[a.length - i]))
-      r += a.coeffs[a.length - i]
+      r = mul!(r, r, b^(reinterpret(Int, a.exps[a.length - i + 1] - a.exps[a.length - i])))
```

Comment:
```suggestion
      r = mul!(r, b^(reinterpret(Int, a.exps[a.length - i + 1] - a.exps[a.length - i])))
```
you can make use of the shorthand if the first and second arg are identical (at most of the changed places)